report/topology.go

```diff
@@ -78,8 +79,9 @@ func (n Nodes) Merge(other Nodes) Nodes {
 // given node in a given topology, along with the edges emanating from the
 // node and metadata about those edges.
 type Node struct {
-	Metadata  `json:"metadata,omitempty"`
-	Counters  `json:"counters,omitempty"`
+	Metadata  Metadata      `json:"metadata,omitempty"`
+	Counters  Counters      `json:"counters,omitempty"`
+	Sets      Sets          `json:"sets,omitempty"`
```

Comment:
I would put Sets inside Metadata (it's just a different way to structure metadata). Also, I wonder If we can avoid the heterogeneous representation (maps and sets) by using interfaces with getters instead of plain strings.